src/main/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -763,7 +801,7 @@ private void assertIntervalsOptimized(final QueryInterval[] intervals) {
     /**
      * Iterate over the SAMRecords defined by the sections of the file described in the ctor argument.
      */
-    private class BAMFileIndexIterator extends BAMFileIterator {
+    public class BAMFileIndexIterator extends BAMFileIterator {
```

Comment:
@droazen can you use the review feature so I don't get a notification per comment?